cilium-cli/connectivity/sniff/filters.go

```diff
@@ -3,13 +3,38 @@
 
 package sniff
 
-// TunnelFilter is a tcpdump filter which captures encapsulated packets.
+import (
+	"fmt"
+
+	"github.com/cilium/cilium/cilium-cli/connectivity/check"
+	"github.com/cilium/cilium/cilium-cli/utils/features"
+)
+
+// GetTunnelFilter returns a tcpdump filter which captures encapsulated packets.
 //
 // Some explanations:
-//   - "udp[8:2] = 0x0800" compares the first two bytes of an UDP payload
-//     against VXLAN commonly used flags. In addition we check against
-//     the default Cilium's VXLAN port (8472).
-//   - To catch Geneve traffic we cannot use the "geneve" filter, as it shifts
-//     offset of a filtered packet, which invalidates a filter matching on the
-//     outer headers. Thus this poor UDP/6081 check.
-const TunnelFilter = "(udp and (udp[8:2] = 0x0800 or dst port 8472 or dst port 6081))"
+//   - VXLAN: "udp[8:2] = 0x0800" compares the first two bytes of an UDP payload
+//     against VXLAN commonly used flags. In addition we check against the current tunnel-port.
+//   - Geneve: we cannot use the "geneve" filter, as it shifts offset of a filtered packet,
+//     which invalidates a filter matching on the outer headers. Thus we only check against
+//     the current tunnel-port.
+func GetTunnelFilter(ct *check.ConnectivityTest) (string, error) {
+	tunnelProtocol := ct.Features[features.Tunnel]
+	tunnelPort := ct.Features[features.TunnelPort]
+
+	if !tunnelProtocol.Enabled {
+		return "", nil
```

Comment:
Shouldn't this return an error instead? AFAICS, it should never happen, but were it to happen, we don't want to return "" or tcpdump will reject the filter.